doc/htm.md

```diff
@@ -0,0 +1,37 @@
+HTM Indexing
+============
+
+Overview                        {#htm-overview}
+========
```

Comment:
I'll try, but IIRC I had trouble getting doxygen to do sensible things with this (those weird braced anchors after the header are necessary in order to be able to link to sections from within code comments). Another thing I should point out it that the top-level README.md is also used by doxygen as the front-page for the docs it generates. The cost is that one of those braced anchors is visible when github renders the markdown, which I thought was acceptable. Feel free to disagree :)